torch/jit/_check.py

```
@@ -0,0 +1,214 @@
+                return
+            if node.value.elts:
+                return
+        elif ann_type == "Dict":
+            # Assigning `{}` to a `Dict` type gives you an AnnAssign
+            # Node where value=Dict(keys=[], values=[])
+            if not isinstance(node.value, ast.Dict):
+                return
+            if node.value.keys:
+                return
+        elif ann_type == "Optional":
+            # Assigning `None` to an `Optional` type gives you an
+            # AnnAssign Node where value=Constant(value=None, kind=None)
+            # or, in Python <3.8, value=NameConstant(value=None)
+            if (not self.using_deprecated_ast
+                    and not isinstance(node.value, ast.Constant)):
+                return
+            if (self.using_deprecated_ast
+                    and not isinstance(node.value, ast.NameConstant)):
+                return
+            if node.value.value:
+                return
+
+        raise RuntimeError("The TorchScript type system doesn't support"
+                           " instance-level annotations on empty"
+                           " non-base types in `__init__`. Instead, "
+                           " either 1) use a type annotation in the "
+                           "class body, or 2) wrap the type in "
+                           "`torch.jit.Attribute`.")
+
+    def visit_Call(self, node):
+        """
+        Visit a Call node in an ``nn.Module``'s ``__init__``
+        method and determine if it's ``torch.jit.annotate``. If so,
+        see if it conforms to our attribute annotation rules.
+        """
+        # If this isn't a call to `torch.jit.annotate`
+        try:
+            if (node.func.value.value.id != "torch"
+                    or node.func.value.attr != "jit"
+                    or node.func.attr != "annotate"):
+                try:
+                    self.generic_visit(node)
+                except RuntimeError:
```

Shouldn't we just call `self.generic_visit` and let the error propagate naturally? What is the purpose of the try-except-raise?